pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1180,17 +1377,19 @@ var (
   kind: Namespace
   metadata:
     labels:
-      provider: codeready-toolchain
-      project: codeready-toolchain
+      toolchain.dev.openshift.com/provider: codeready-toolchain
+      toolchain.dev.openshift.com/project: codeready-toolchain
+      toolchain.dev.openshift.com/owner: ${USERNAME}
```

Comment:
the template objects shouldn't contain the `project` and `owner` toolchain labels. In addition, we don't have `toolchain.dev.openshift.com/project` label key at all - correct me if I'm wrong.

Anyway, for now the only toolchain label that can be part of the template is the `toolchain.dev.openshift.com/provider`. The rest should be added to the object by the NSTemplateSet controller. In other words, if the labels were specified here then the tests would miss their purpose.

Reply:
good point, let me remove these labels.

Reply:
the provider label won't be specified in the template for long though..

Reply:
> Also, could you please double check the test coverage? It seems that there is some room for coverage improvement.

I've added some tests in e64c93b. There are still some cases of errors that could be covered by mocking the client. But could we address that in a separate PR?

Reply:
this was addressed in https://github.com/codeready-toolchain/member-operator/pull/143/commits/e462a7d53e20a83b5184634bf3a859f96880e986